bifromq-inbox/bifromq-inbox-server/src/test/java/com/baidu/bifromq/inbox/server/InboxWriterPipelineTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2025. The BifroMQ Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+package com.baidu.bifromq.inbox.server;
+
+import com.baidu.bifromq.baseenv.MemUsage;
+import com.baidu.bifromq.baserpc.RPCContext;
+import com.baidu.bifromq.baserpc.metrics.IRPCMeter;
+import com.baidu.bifromq.baserpc.metrics.RPCMetric;
+import com.baidu.bifromq.inbox.rpc.proto.SendReply;
+import com.baidu.bifromq.inbox.rpc.proto.SendRequest;
+import com.baidu.bifromq.inbox.util.PipelineUtil;
+import com.baidu.bifromq.plugin.subbroker.DeliveryReply;
+import com.baidu.bifromq.plugin.subbroker.DeliveryResult;
+import com.baidu.bifromq.plugin.subbroker.DeliveryResults;
+import com.baidu.bifromq.sysprops.props.IngressSlowDownDirectMemoryUsage;
+import com.baidu.bifromq.sysprops.props.IngressSlowDownHeapMemoryUsage;
+import com.baidu.bifromq.type.MatchInfo;
+import io.grpc.Context;
+import io.grpc.stub.ServerCallStreamObserver;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static com.baidu.bifromq.inbox.records.ScopedInbox.receiverId;
```

Comment:
Pls reformat your code according to the project's [checkstyle](https://github.com/bifromqio/bifromq/blob/main/checkstyle.xml)